salt/modules/beacons.py

```diff
@@ -361,7 +382,7 @@ def save():
     beacons = list_(return_yaml=False, include_pillar=False)
 
     # move this file into an configurable opt
-    sfn = os.path.join(__opts__['config_dir'],
+    sfn = os.path.join(os.path.dirname(__opts__['conf_file']),
```

Comment:
Is this a bug fix? I know I was looking at the config dir thing recently, because if no `minion` file exists there then we override it with the default :)

If this is a pattern that we should be following everywhere, FWIW there are a few other locations using `config_dir`:

```
 %  ~/p/salt  ╍╍ rfc-changing-requisites  salt  ag opts__..config_dir .
wheel/config.py
75:    dir_path = os.path.join(__opts__['config_dir'],

cloud/clouds/lxc.py
68:        default_dir = __opts__['config_dir'],
79:        default_dir = __opts__['config_dir'],

modules/beacons.py
364:    sfn = os.path.join(__opts__['config_dir'],

modules/schedule.py
837:    sfn = '{0}/{1}/schedule.conf'.format(__opts__['config_dir'], os.path.dirname(__opts__['default_include']))
```

Reply:
I believe `config_dir` could be an artifact from code that we dropped.
`config_dir` is defined when the CLI scripts run, but that value is not "transposed" to the loaded config.
This change is basically removing those artifacts, but apparently I missed some.